File: MultiplesOfaNumber/multiplesofanumber.js
/*Sample code to read in test cases:*/
var fs  = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line !== "") {
        var nums = line.split(','),
            x = parseInt(nums[0], 10),
            n = parseInt(nums[1], 10),
            a = n,
            i = 2;
        while(a < x){
            a += n;
        }
        console.log(a);
    }
});

File: MultiplesOfaNumber/multiplesofanumber.clj
(defn moan [anchor contender]
  (loop [a anchor c contender]
    (if (<= a c)
      c
      (recur a (+ c c))
    )
  )
)

(defn tint [x] (Integer/parseInt x))

(with-open [rdr (clojure.java.io/reader (first *command-line-args*))]
  ; Read each line ignoring empty ones
  (doseq [line (remove empty? (line-seq rdr))]
    (let [nums (clojure.string/split line #",")]
      (let [[a c] nums]
        (println (moan (tint a) (tint c)))
      )
    )
  )
)